src/documentation/io/authoring-overview.md

```diff
@@ -10,35 +10,80 @@ permalink: /documentation/io/authoring-overview/
+
+As you work on your I/O Transform, be aware that the Beam community is excited to help those building new I/O Transforms and that there are many examples and helper classes.
 
-## Example I/O Transforms
-TODO
 
 ## Suggested steps for implementers
-TODO
+1. Check out this guide and come up with your design. If you'd like, you can email the [Beam dev mailing list]({{ site.baseurl }}/get-started/support) with any questions you might have. It's good to check there to see if anyone else is working on the same I/O Transform.
+2. If you are planning to contribute your I/O transform to the Beam community, you'll be going through the normal Beam contribution life cycle - see the [Apache Beam Contribution Guide]({{ site.baseurl }}/contribute/contribution-guide/) for more details.
+3. As you're working on your IO transform, see the [PTransform Style Guide]({{ site.baseurl }}/contribute/ptransform-style-guide/) for specific information about writing I/O Transforms.
+
 
 ## Read transforms
-TODO
+Read transforms take data from outside of the Beam pipeline and produce PCollections of data.
+
+For data stores or files types where the data can be read in parallel, you can think of the process as a mini-pipeline. This often consists of two steps:
+1. Splitting the data into parts to be read in parallel
+2. Reading from each of those parts
+
+Each of those steps will be a `ParDo`, with a `GroupByKey` in between. The `GroupByKey` is an implementation detail, but for most runners it allows the runner to use different numbers of workers for:
+* Determining how to split up the data to be read into chunks - this will likely occur on very few workers
+* Reading - will likely benefit from more workers
+
+The `GroupByKey` will also allow Dynamic Work Rebalancing to occur (on supported runners).
+
+Here are some examples of read transform implementations that use the "reading as a mini-pipeline" model when data can be read in parallel:
+* **Reading from a file glob** - For example reading all files in "~/data/**"
+  * Get File Paths `ParDo`: As input, take in a file glob. Emit a PCollection of strings, each of which is a file path.
+  * Reading `ParDo`: Given the PCollection of file paths, read each one, emitting a PCollection of records.
+* **Reading from a NoSQL Database** (eg Apache HBase) - these databases often allow reading from ranges in parallel.
+  * Determine Key Ranges `ParDo`: As input, receive connection information for the database and the key range to read from. Output a PCollection of key ranges that can be read in parallel efficiently.
+  * Read Key Range `ParDo`: Given the PCollection of key ranges, read the key range, emitting a PCollection of records.
+
+For data stores or files where reading cannot occur in parallel, reading is a simple task that can be accomplished with a single `ParDo`+`GroupByKey`. For example:
```

Comment:
correct. That's a good detail to put in another document or to pick up from examples (linked to in the java overview.) This overview noticeably does not go into that type of detail on any of the programming topics.